services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/operations/PutItemOperation.java

```diff
@@ -99,18 +101,31 @@ public PutItemRequest generateRequest(TableSchema<T> tableSchema,
                                                               .tableName(operationContext.tableName())
                                                               .item(itemMap);
 
+        if (request.left().isPresent()) {
+            requestBuilder = addPlainPutItemParameters(requestBuilder, request.left().get());
+        }
```

Comment:
Nit:
```java
request.left().ifPresent(r -> addPlainPutItemParameters(requestBuilder, r));
```

Reply:
Tried it originally, but java complains that `requestBuilder` isn't effectively final ☹️